aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -14,6 +14,100 @@
 namespace at {
 namespace native {
 
+namespace {
+// dense_to_sparse_{csr,bsr,csc,bsc} common helpers
+
+// Preparation fo the N-D dense -> sparse compressed conversion.
+// The N-D input is converted to 3-D (single batch dim) where we check that the
+// product of batch dims is nonzero and for each batch the sparse matrix
+// contained within has the same number of non-zero elements.
+// The batches are joined along the compressed axis. The generation of indices
+// for this matrix can be performed in a single step followed by a single step
+// conversion to restore the batch dimension.
+void dense_to_sparse_compressed_prepare_check_mask_values_batched(
+    Layout const& target_layout,
+    Tensor& values,
+    Tensor& mask,
+    const int64_t& n_batch_dim) {
+  if (n_batch_dim > 1) {
+    // For inputs with more than 1 batch dim we flatten them out.
+    // Input shape (b0, b1 ..., bn, r, c) -> (b, r ,c)
+    values = values.flatten(0, n_batch_dim - 1);
+    mask = mask.flatten(0, n_batch_dim - 1);
+  }
+
+  TORCH_CHECK(
+      mask.size(0) > 0,
+      "to_sparse_",
+      sparse_csr::layoutToString(target_layout, false, true),
```

Comment:
nit: we could document the function and explain what `target_layout` represents and how it is connected to the last `false, true` arguments of `layoutToString`.